api/tests/mocha/services/xforms/markdown/xform.xml

```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0"?>
+<h:html xmlns="http://www.w3.org/2002/xforms" xmlns:ev="http://www.w3.org/2001/xml-events" xmlns:h="http://www.w3.org/1999/xhtml" xmlns:jr="http://openrosa.org/javarosa" xmlns:orx="http://openrosa.org/xforms" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+  <h:head>
+    <h:title>Note Demo</h:title>
+    <model>
+      <itext>
+        <translation lang="en">
+          <text id="name_note:label">
+            <value>
+# Hello <output value="/notedata/name"/>!
+</value>
+          </text>
+          <text id="lmp_note:label">
+            <value><output value="/notedata/name"/> is currently &lt;span style=&quot;font-family:monospace&quot;&gt;12 weeks&lt;/span&gt; pregnant.</value>
+          </text>
+          <text id="patient_details:label">
+            <value>Remind woman of the **6 STEPS TO A HEALTHY PREGNANCY:**
+
+             &lt;ol style=&quot;margin-left:1em; list-style: decimal inside;&quot;&gt;&lt;li&gt;Attend regular ANC visits&lt;/li&gt;&lt;li&gt;Sleep under a treated net **every** night&lt;/li&gt;&lt;li&gt;<output value="/notedata/name"/> is in her **first trimester**. Remind her to take supplements:&lt;ul style=&quot;margin-left:1em; list-style: disc inside;&quot;&gt;&lt;li&gt;_Take Iron Folate daily_&lt;/li&gt;&lt;/ul&gt;&lt;/li&gt;&lt;li&gt;Eat well: Eat more often than usual and eat a variety of foods to give you strength and to help your baby grow&lt;/li&gt;&lt;li&gt;Deliver your child at a health center&lt;/li&gt;&lt;li&gt;Breastfeed the baby **immediately** after birth&lt;/ol&gt;</value>
+          </text>
+          <text id="referrals:label">
+            <value>Referrals&lt;I class="fa fa-hospital-o"&gt;&lt;/i&gt;</value>
+          </text>
+          <text id="submit_note:label">
+            <value>&lt;h4 style="text-align:center;"&gt;Click the Submit button at the bottom of the form.&lt;/h4&gt;</value>
+          </text>
+        </translation>
+      </itext>
+      <instance>
+        <notedata delimiter="#" id="note_demo">
+          <name/>
+          <name_note/>
+          <meta>
+            <instanceID/>
+          </meta>
+        </notedata>
+      </instance>
+      <bind nodeset="/notedata/name" type="string"/>
+      <bind nodeset="/notedata/name_note" readonly="true()" type="string"/>
```

Comment:
So, I considered adding a readonly note in one of these test forms to verify the xslt.  The only issue is that AFIK the unit tests just [mock out the call to `xsltproc`](https://github.com/medic/cht-core/blob/9a3eeef426691ebb2f4af5be47de50501791f9d8/api/tests/mocha/services/generate-xform.spec.js#L69) so that  whatever is in the `form.html` gets returned (and does not actually use the xslt to convert the xml to html).  I assume this was to avoid having our unit tests dependent on `xsltproc`.  

One interesting alternative would be to add [`libxslt`](https://www.npmjs.com/package/libxslt) as a devDependency and then let the tests use that to do the conversion.  Would not be an exact one-for-one test of the actual code that uses `xsltproc`, but might be better than nothing (which is pretty much what we have now...).  Let me know what you think!

(Another thing to note is that this issue with testing `xsltproc` functionality should go away once we switch fully over to `enketo-transformer`.)

Reply:
Oh, I see. Then we should definitely have some integration or e2e test. I think e2e test is easiest.

Reply:
Okay, so I think this is the first time I have written one of these e2e tests from scratch, so let me know if I did something wrong here: https://github.com/medic/cht-core/blob/6345-enketo-uplift/tests/e2e/api/services/generate-xform.wdio-spec.js

Seems like the test works since it passes when I run it, but if I remove the changes to the xsl file and run the test it fails...

Reply:
The test is great! Awesome job!